src/cli/cli-report.c

```diff
@@ -738,16 +738,16 @@ static int run_event_chain_real(struct run_event_state *run_state,
                                 GList                  *chain,
                                 int                     interactive)
 {
-    struct logging_state l_state;
+    g_autofree struct logging_state *l_state = g_malloc0(sizeof(struct logging_state));
     int retval = 0;
 
     /* Blergh. */
     run_state->logging_callback = do_log2;
-    run_state->logging_param = &l_state;
+    run_state->logging_param = l_state;
```

Comment:
Doesn't this lead to a dangling pointer once `l_state` is autofreed?

Reply:
Good catch :D , that would be nasty use-after-free. It should be freed in the free_run_event_state not by autofree.